syntax = "proto3";

package hermeneutic.grpc;

message SubscribeRequest {
  string symbol = 1;
}

message AggregatedQuote {
  string price = 1;
  string quantity = 2;
}

message AggregatedBook {
  AggregatedQuote best_bid = 1;
  AggregatedQuote best_ask = 2;
  uint32 exchange_count = 3;
  int64 timestamp_unix_millis = 4;
  repeated AggregatedQuote bid_levels = 5;
  repeated AggregatedQuote ask_levels = 6;
  int64 last_feed_timestamp_ns = 7;
  int64 last_local_timestamp_ns = 8;
  int64 min_feed_timestamp_ns = 9;
  int64 max_feed_timestamp_ns = 10;
  int64 min_local_timestamp_ns = 11;
  int64 max_local_timestamp_ns = 12;
}

service AggregatorService {
  rpc StreamBooks(SubscribeRequest) returns (stream AggregatedBook);
}
